renderdoc/api/replay/resourceid.h

```diff
@@ -24,6 +24,8 @@
 
 #pragma once
 
+#include <functional>
```

Comment:
It's not for that, it's because `ResourceId` has a `friend struct std::hash<ResourceId>` declaration in it, but `std::hash` hasn't been declared at this point.  It would be possible to forward declare everything and then implement the `std::hash<ResourceId>()` operator in a source file after including one of the appropriate std headers - there would be a performance hit for this though.

A medium solution would be to move the include to above the `ResourceId` include in `gpu_address_range_tracker.h`, it's still commonly included (as `vk_core.h` and `d3d12_manager.h` include it) though.

Reply:
Oh I see, then that's my mistake - I thought I'd expunged most of these heavyweight STL headers but some are still needed. I guess this wasn't present before because usually code doesn't include just `resourceid.h` on its own. This is fine then.